Controller/ApiController.php

```diff
@@ -88,6 +98,64 @@ private function treatFormRequest(Request $request, string $apiName)
         return $body;
     }
 
+
+    private function validateHashcash(Request $request, string $csrfId, int $hashcashLevel, string $hashAlgo)
+    {
+        $hashcash = $request->headers->get('X-Hashcash');
+        if ($hashcash === null) {
+            throw new AccessDeniedHttpException('Unrecognized user');
+        }
+
+        $tokens = explode('|', $hashcash);
```

Comment:
Would be better to handle this in an object? To centralize the processing of the token (makes it useable for the formbundle)

Reply:
Done